examples/enrichment/Examples.Enrichment/MyTraceEnricher.cs

```diff
@@ -0,0 +1,27 @@
+// <copyright file="MyTraceEnricher.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.Extensions.Enrichment;
+
+namespace Examples.Enrichment;
+
+internal class MyTraceEnricher : TraceEnricher
+{
+    public override void Enrich(TraceEnrichmentBag bag)
+    {
+        bag.Add("my key", "my value");
```

Comment:
Should we provide another `TraceEnricher` which does something more interesting? As like an advanced example? Thinking inject something into ctor (like "MyService") which pushes some contextual information as a tag. The current example really could be accomplished just using `Resource` so it isn't highlighting the best aspects of this library 😄

What I mean is, for static data, you could accomplish this with...

```csharp
    public static IServiceCollection AddMyEnricher(this IServiceCollection services)
    {
        return services.ConfigureOpenTelemetryTracerProvider((sp, builder) =>
           builder.ConfigureResource(resourceBuilder => resourceBuilder.AddAttributes(new Dictionary<string, object>
           {
               ["my key"] = "my value",
           })));
    }
```

Reply:
Updated